govcd/provider_vdc.go

```diff
@@ -182,3 +185,405 @@ func getProviderVdcByName(vcdClient *VCDClient, providerVdcName string, extended
 	}
 	return vcdClient.GetProviderVdcByHref(foundProviderVdcs.Results.VMWProviderVdcRecord[0].HREF)
 }
+
+// CreateProviderVdc creates a new provider VDC using the passed parameters
+func (vcdClient *VCDClient) CreateProviderVdc(params *types.ProviderVdcCreation) (*ProviderVdcExtended, error) {
+	if !vcdClient.Client.IsSysAdmin {
+		return nil, fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if params.Name == "" {
+		return nil, fmt.Errorf("a non-empty name is needed to create a provider VDC")
+	}
+	if len(params.ResourcePoolRefs.VimObjectRef) == 0 {
+		return nil, fmt.Errorf("resource pool is needed to create a provider VDC")
+	}
+	if len(params.StorageProfile) == 0 {
+		return nil, fmt.Errorf("storage profile is needed to create a provider VDC")
+	}
+	if params.VimServer == nil {
+		return nil, fmt.Errorf("vim server is needed to create a provider VDC")
+	}
+	pvdcCreateHREF := vcdClient.Client.VCDHREF
+	pvdcCreateHREF.Path += "/admin/extension/providervdcsparams"
+
+	resp, err := vcdClient.Client.ExecuteJsonRequest(pvdcCreateHREF.String(), http.MethodPost, params, "error creating provider VDC: %s")
+	if err != nil {
+		return nil, err
+	}
+
+	body, _ := io.ReadAll(resp.Body)
+	util.ProcessResponseOutput(util.CallFuncName(), resp, string(body))
```

Comment:
Is this function already called inside `ExecuteJsonRequest`?

 https://github.com/vmware/go-vcloud-director/pull/580/files#diff-db6a2159c7e76cc15aae02bb4455af711bd8fc33263e1fb73d50be2a344a0c0dR60

`body` also must be closed, but maybe all of this can fit into `ExecuteJsonRequest` itself (as in my original comment in function)

Reply:
Added body closure to all usages of executeJsonRequest. Also added a note in the function documentation about closing responsibility